netty/src/main/java/io/grpc/transport/netty/NettyServerHandler.java

```diff
@@ -84,13 +84,25 @@
   private Throwable connectionError;
   private ChannelHandlerContext ctx;
   private boolean teWarningLogged;
+  private int connectionWindowSize;
 
   NettyServerHandler(ServerTransportListener transportListener,
       Http2Connection connection,
       Http2FrameReader frameReader,
       Http2FrameWriter frameWriter,
-      int maxStreams) {
+      int maxStreams,
+      int connectionWindowSize,
+      int streamWindowSize) {
     super(connection, frameReader, frameWriter, new LazyFrameListener());
+    Preconditions.checkArgument(connectionWindowSize > 0, "connectionWindowSize must be positive");
+    this.connectionWindowSize =
+        connectionWindowSize == streamWindowSize ? -1 : connectionWindowSize;
```

Comment:
haha ... no, feel free to remove it.  Just remove it from the client too so it's consistent.